pkg/engine/lifecycletest/hide_diffs_test.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+func TestHideDiffs_Plain(t *testing.T) {
+	t.Parallel()
+	testHideDiffs(t, false)
+}
+
+func TestHideDiffs_Detailed(t *testing.T) {
+	t.Parallel()
+	testHideDiffs(t, true)
+}
+
+func testHideDiffs(t *testing.T, detailedDiff bool) {
+	propValue := resource.NewProperty("a")
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				DiffF: func(_ context.Context, req plugin.DiffRequest) (plugin.DiffResult, error) {
+					if req.OldInputs.DeepEquals(req.NewInputs) {
+						return plugin.DiffResult{
+							Changes: plugin.DiffNone,
+						}, nil
+					}
+
+					p := plugin.DiffResult{Changes: plugin.DiffSome}
+					if detailedDiff {
+						p.DetailedDiff = plugin.NewDetailedDiffFromObjectDiff(req.OldInputs.Diff(req.NewInputs), true)
+					}
+
+					check := func(key resource.PropertyKey) {
+						if !req.OldInputs[key].DeepEquals(req.NewInputs[key]) {
+							p.ChangedKeys = append(p.ChangedKeys, key)
+						}
+					}
+
+					check("scalar")
+					check("array")
+					check("map")
+
+					return p, nil
+				},
+				CreateF: func(_ context.Context, req plugin.CreateRequest) (plugin.CreateResponse, error) {
+					return plugin.CreateResponse{
+						ID:         "created-id",
+						Properties: req.Properties,
+						Status:     resource.StatusOK,
+					}, nil
+				},
+				UpdateF: func(_ context.Context, req plugin.UpdateRequest) (plugin.UpdateResponse, error) {
+					return plugin.UpdateResponse{
+						Properties: req.NewInputs,
+						Status:     resource.StatusOK,
+					}, nil
+				},
+			}, nil
+		}, deploytest.WithGrpc),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		_, err := monitor.RegisterResource("pkgA:m:typA", "resA", true, deploytest.ResourceOptions{
+			Inputs: resource.PropertyMap{
+				"scalar": propValue,
+				"array": resource.NewProperty([]resource.PropertyValue{
+					propValue,
+				}),
+				"map": resource.NewProperty(resource.PropertyMap{
+					"k": propValue,
+				}),
+			},
+			HideDiffs: []resource.PropertyPath{
+				{"array"},
+				{"map"},
+				{"scalar"},
+			},
+		})
+		require.NoError(t, err)
+		return nil
+	})
+	hostF := deploytest.NewPluginHostF(nil, nil, programF, loaders...)
+
+	p := &lt.TestPlan{
+		Options: lt.TestUpdateOptions{
+			T:                t,
+			HostF:            hostF,
+			SkipDisplayTests: false,
+		},
+		Steps: []lt.TestStep{
+			{
+				Op: engine.Update,
+				Validate: func(
+					project workspace.Project, target deploy.Target, entries engine.JournalEntries,
+					events []engine.Event, err error,
+				) error {
+					lt.AssertDisplay(t, events, filepath.Join("testdata", "output", t.Name()))
+					return nil
+				},
```

Comment:
I don't think we need this, `p.Run` should already call `AssertDisplay`.  Maybe worth a comment that we're running this tests just for the display portion of the test.